### ISSUE TITLE
Breaking: Rewrite

### ISSUE THREAD

**Issue by jorgebucaran** [jorgebucaran/nvm.fish#75]
I've neglected this project for a long time. Now, I'd like to rewrite it (from scratch), fix all the issues that have collected over the years and rename it (_fnm_ has poor discoverability/SEO). 100% backward compatibility is not a goal. 

I'll be updating this issue to clarify exactly what will change, what features will be added or deprecated, what breaking changes will be introduced, and everything else that you need to know about.

**Comment by hybridvision** [comment]
This sounds great! Nothing to add in terms of usage since it is very well thought-out already.

For the naming, I think `n` is nice and hopefully won't conflict with people's existing aliases. My other suggestion is: `nv`

Thanks for your work on this @jorgebucaran!

**Comment by saintwinkle** [comment]
Vote for `n`, but also like the old `fnm` 😂

And consider to add `n ls-remote` just as `nvm`? Command `fnm ls` at moment will output a very long list, that I can hardly distinguish my local versions from remote.

**Comment by jorgebucaran** [comment]
@hybridvision Awesome!
@saintwinkle Thanks. True, can't filter the list with the current `ls`, but future `ls` could. See #76.

```
n ls 10.1
10.1.0
10.10.0
...
```

**Comment by jorgebucaran** [comment]
Here is what the `n ls` output looks like:

```console
0.10.0
0.10.1
0.10.2
0.10.3
0.10.4
0.10.5
0.10.6
0.10.7
0.10.8
0.10.9
0.10.10
0.10.11
0.10.12
0.10.13
0.10.14
0.10.15
0.10.16
0.10.17
0.10.18
0.10.19
0.10.20
0.10.21
0.10.22
0.10.23
0.10.24
0.10.25
0.10.26
0.10.27
0.10.28
0.10.29
0.10.30
0.10.31
0.10.32
0.10.33
0.10.34
0.10.35
0.10.36
0.10.37
0.10.38
0.10.39
0.10.40
0.10.41
0.10.42
0.10.43
0.10.44
0.10.45
0.10.46
0.10.47
0.10.48 (0.10)
0.11.0
0.11.1
0.11.2
0.11.3
0.11.4
0.11.5
0.11.6
0.11.7
0.11.8
0.11.9
0.11.10
0.11.11
0.11.12
0.11.13
0.11.14
0.11.15
0.11.16
0.12.0
0.12.1
0.12.2
0.12.3
0.12.4
0.12.5
0.12.6
0.12.7
0.12.8
0.12.9
0.12.10
0.12.11
0.12.12
0.12.13
0.12.14
0.12.15
0.12.16
0.12.17
0.12.18 (0.12)
4.0.0
4.1.0
4.1.1
4.1.2
4.2.0
4.2.1
4.2.2
4.2.3
4.2.4
4.2.5
4.2.6
4.3.0
4.3.1
4.3.2
4.4.0
4.4.1
4.4.2
4.4.3
4.4.4
4.4.5
4.4.6
4.4.7
4.5.0
4.6.0
4.6.1
4.6.2
4.7.0
4.7.1
4.7.2
4.7.3
4.8.0
4.8.1
4.8.2
4.8.3
4.8.4
4.8.5
4.8.6
4.8.7
4.9.0
4.9.1 (argon)
5.0.0
5.1.0
5.1.1
5.2.0
5.3.0
5.4.0
5.4.1
5.5.0
5.6.0
5.7.0
5.7.1
5.8.0
5.9.0
5.9.1
5.10.0
5.10.1
5.11.0
5.11.1
5.12.0  (5)
6.0.0
6.1.0
6.2.0
6.2.1
6.2.2
6.3.0
6.3.1
6.4.0
6.5.0
6.6.0
6.7.0
6.8.0
6.8.1
6.9.0
6.9.1
6.9.2
6.9.3
6.9.4
6.9.5
6.10.0
6.10.1
6.10.2
6.10.3
6.11.0
6.11.1
6.11.2
6.11.3
6.11.4
6.11.5
6.12.0
6.12.1
6.12.2
6.12.3
6.13.0
6.13.1
6.14.0
6.14.1
6.14.2
6.14.3
6.14.4
6.15.0
6.15.1
6.16.0  (boron)
7.0.0
7.1.0
7.2.0
7.2.1
7.3.0
7.4.0
7.5.0
7.6.0
7.7.0
7.7.1
7.7.2
7.7.3
7.7.4
7.8.0
7.9.0
7.10.0
7.10.1  (7)
8.0.0
8.1.0
8.1.1
8.1.2
8.1.3
8.1.4
8.2.0
8.2.1
8.3.0
8.4.0
8.5.0
8.6.0
8.7.0
8.8.0
8.8.1
8.9.0
8.9.1
8.9.2
8.9.3
8.9.4
8.10.0
8.11.0
8.11.1
8.11.2
8.11.3
8.11.4
8.12.0
8.13.0
8.14.0
8.14.1
8.15.0  (carbon)
9.0.0
9.1.0
9.2.0
9.2.1
9.3.0
9.4.0
9.5.0
9.6.0
9.6.1
9.7.0
9.7.1
9.8.0
9.9.0
9.10.0
9.10.1
9.11.0
9.11.1
9.11.2  (9)
10.0.0
10.1.0
10.2.0
10.2.1
10.3.0
10.4.0
10.4.1
10.5.0
10.6.0
10.7.0
10.8.0
10.9.0
10.10.0
10.11.0
10.12.0
10.13.0
10.14.0
10.14.1
10.14.2
10.15.0 (lts/dubnium)
11.0.0  
11.1.0
11.2.0  <-
11.3.0
11.4.0
11.5.0
11.6.0  (latest)
```

The legend should be self-explanatory. The `<-` indicates the version that I am currently using. 

Next to some releases in parentheses is the main version number, codename and in the case of the most recent long term release (lts), that is also shown.

For example:

```
10.15.0 (lts/dubnium)
```

Means you can install 10.15.0 using either `n lts` or `n dubnium`. Incidentally, `n 10` works too. You can use the major and minor to install the latest patch as well: `n 10.14` will install `10.14.2`.

To install the latest release you can use `n latest`, which should install `11.6.0` at the time of this writing.

**Comment by shadowspawn** [comment]
(Disclaimer: I am an interested observer, not an `fnm` user.)

I reworked the `ls` command in [nvh](https://github.com/JohnRGee/nvh), my feature fork of [tj/n](
https://github.com/tj/n), and covered some similar ground. Offered for interest:

1) I renamed to `ls-remote` (alias `lsr`) like in [nvm](https://github.com/creationix/nvm#usage).

2) I implemented single version lookups for lts and latest like:

    $ nvh lsr lts
    v10.15.0

3) I used codenames and partial numeric versions to limit output. In addition I implemented an idea from [tj/n/#383](https://github.com/tj/n/issues/383#issuecomment-404411323) to limit number displayed by default, so combined:

```sh
$ nvh lsr dubnium
Listing remote... Displaying 20 matches (use --all to see all).
v10.15.0
v10.14.2
v10.14.1
v10.14.0
v10.13.0
```

**Comment by jorgebucaran** [comment]
@JohnRGee Thanks for your input. For the record, here are my thoughts on that for this package.

1. Only one `ls` is better. 
2. I like that. Another filter I'm considering to add is `stable` (to show only stable releases -- even numbered major versions). 
3. I find more useful the ability to install a binary from a codename, which we'll have, of course. 

Adding an `--all` flag to `ls` is a good idea. I'll keep it in mind.

**Comment by lukebennett88** [comment]
Much better name

**Comment by lalitkapoor** [comment]
There is another package called fish-nvm as well (https://github.com/FabioAntunes/fish-nvm). So that might get a little confusing.

**Comment by jorgebucaran** [comment]
@lalitkapoor Are you planning to use both @FabioAntunes's nvm and my nvm? That would be confusing.

**Comment by jorgebucaran** [comment]
Yeah, I see. A different name would allow you to use this package and Fabio's nvm wrapper too. I'll try to think of other names, but I am running out of ideas and I don't want to use another abbreviation. If I can't find a better name, I'll stick to `fish-nvm`. If you have any ideas, please suggest them!

**Comment by jorgebucaran** [comment]
@lalitkapoor Yeah, I see. A different name would allow you to use this package and Fabio's nvm wrapper too. I'll try to think of other names, but I am running out of ideas and I don't want to use another abbreviation. If I can't find a better name, I'll stick to `fish-nvm`. If you have any ideas, please suggest them!

**Comment by jorgebucaran** [comment]
Ideas I considered:

- **fish-chnode**: change node; better than more acronym nonsense
- **fish-mvn**: move node (also nvm spelled backwards); best acronym I could find that didn't suck
- **fish-vertex**: another name for node in grapth theory and contains _ver_ (short for version) in the name; more memorable than an acronym
- **fish-nodeverse**: portmanteau of node and universe; contains the _ver_ in the name too, but a bit too long

fish-nvm is still winning though.

**Comment by devel787** [comment]
Maybe 'fish-nvmgr'.

**Comment by jorgebucaran** [comment]
Thanks, but no acronyms (unless it's `nvm` that is—the only acronym that works well here).

**Comment by lalitkapoor** [comment]
@jorgebucaran What about just sticking with fnm? fish node manager still sounds great to me! :) I'm sorry I don't know the why behind the decision to rename. Can you share more? Thanks!

**Comment by Pajk** [comment]
@jorgebucaran *fnm* is ok but but it's true that every time I want to use it I have to think for a moment and remind myself it's **f**ish **n**ode **m**anager.

I have one feature request - `fnm nightly`!

Thanks for maintaining this project.

**Comment by jorgebucaran** [comment]
@Pajk <<https://nodejs.org/dist>> has no `nightly`, but you can get this working out of the box using:

```fish
set -g nvm_mirror https://nodejs.org/download
```

Thank you for using it!

**Comment by padurean** [comment]
Is there a migration guide - from previous (fnm) version to this one (fish-nvm)? Or is it enough if i just remove fnm from fisher and add fish-nvm?

**Comment by jorgebucaran** [comment]
@padurean Yes, that should be enough. Good luck and let me know if anything goes amiss.

**Comment by vascod-nezasa** [comment]
Sorry to bother here on a closed ticket, but was installing the plugin and trying to follow a document I have here to install nv/node/npm and I noticed that the specific thing in this ticket/PR does not exist or work - https://github.com/jorgebucaran/fish-nvm/pull/67/files

For example, doing nvm, it installs the specified node version in a `.nvmrc` in the above directory, but immediately afterwards I need to upgrade the npm package to `npm install npm@(cat ../.npmversion)` which does not put the locally installed `../node_modules/.bin` in the path.

I don't understand much of this node/nvm/npm stuff and was following instructions. Anyway I noticed this bit. Just a heads up ;)
